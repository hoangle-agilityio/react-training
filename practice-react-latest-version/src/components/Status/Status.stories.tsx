// Libs
import * as React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

// Components
import Status from "components/Status";

export default {
  title: "Components/Status",
  component: Status,
} as ComponentMeta<typeof Status>;

const Template: ComponentStory<typeof Status> = (args) => <Status {...args} />;

export const StatusBase = Template.bind({});
StatusBase.args = {
  ...StatusBase.args,
  label: "Status",
  type: "Others",
  styles: {
    bgColor: "blue.200",
    color: "white.100",
  },
};

export const StatusOpen = Template.bind({});
StatusOpen.args = {
  ...StatusOpen.args,
  type: "Open",
};

export const StatusPaid = Template.bind({});
StatusPaid.args = {
  ...StatusPaid.args,
  type: "Paid",
};

export const StatusDue = Template.bind({});
StatusDue.args = {
  ...StatusDue.args,
  type: "Due",
};

export const StatusInactive = Template.bind({});
StatusInactive.args = {
  ...StatusInactive.args,
  type: "Inactive",
};
