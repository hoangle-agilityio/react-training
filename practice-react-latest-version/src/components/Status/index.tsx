// Libs
import { Badge, BadgeProps } from "@chakra-ui/react";
import * as React from "react";
import { memo } from "react";

// Types
import { StatusType } from "types/common";

type Props = {
  type: StatusType;
  label?: string;
  styles?: BadgeProps;
};

const Status = ({ type, label, styles }: Props) => {
  let labelStatus = "";
  let statusStyles: BadgeProps = {};

  switch (type) {
    case "Open":
      labelStatus = "Open";
      statusStyles = {
        bgColor: "background.info",
        color: "icon.info",
      };
      break;
    case "Paid":
      labelStatus = "Paid";
      statusStyles = {
        bgColor: "background.success",
        color: "icon.success",
      };
      break;
    case "Due":
      labelStatus = "Due";
      statusStyles = {
        bgColor: "background.error",
        color: "icon.error",
      };
      break;
    case "Inactive":
      labelStatus = "Inactive";
      statusStyles = {
        bgColor: "background.default",
        color: "icon.default",
      };
      break;
    default:
      labelStatus = label ?? "";
      break;
  }

  return (
    <Badge
      {...styles}
      {...statusStyles}
      borderRadius="10px"
      padding="1px 10px"
      fontFamily="medium"
      fontSize="sm"
      fontWeight="500"
      textTransform="capitalize"
    >
      {labelStatus}
    </Badge>
  );
};

export default memo(Status);
