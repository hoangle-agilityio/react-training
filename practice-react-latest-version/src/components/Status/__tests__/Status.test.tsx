import { render } from "@testing-library/react";
import * as React from "react";
import Status from "..";

const props = {
  label: "Status",
  styles: {
    bgColor: "icon.info",
    color: "brand.100",
  },
};

describe("Status Component", () => {
  test("Should match snapshoot for Status base", () => {
    const dom = render(<Status {...props} type="Others" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Status Open", () => {
    const dom = render(<Status type="Open" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Status Paid", () => {
    const dom = render(<Status type="Paid" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Status Due", () => {
    const dom = render(<Status type="Due" />);

    expect(dom).toMatchSnapshot();
  });

  test("Should match snapshoot for Status Inactive", () => {
    const dom = render(<Status type="Inactive" />);

    expect(dom).toMatchSnapshot();
  });
});
