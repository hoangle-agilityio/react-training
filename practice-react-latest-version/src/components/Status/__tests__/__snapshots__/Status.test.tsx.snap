// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Component Should match snapshoot for Status Due 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="chakra-badge css-r7yqe2"
      >
        Due
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="chakra-badge css-r7yqe2"
    >
      Due
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Status Component Should match snapshoot for Status Inactive 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="chakra-badge css-10nvh0g"
      >
        Inactive
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="chakra-badge css-10nvh0g"
    >
      Inactive
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Status Component Should match snapshoot for Status Open 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="chakra-badge css-18jsln3"
      >
        Open
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="chakra-badge css-18jsln3"
    >
      Open
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Status Component Should match snapshoot for Status Paid 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="chakra-badge css-1dmlmz1"
      >
        Paid
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="chakra-badge css-1dmlmz1"
    >
      Paid
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Status Component Should match snapshoot for Status base 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="chakra-badge css-cl8l8e"
      >
        Status
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="chakra-badge css-cl8l8e"
    >
      Status
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
